### PR TITLE
tests: move SerialConsole tests over to PexpectEngine/ConsoleBase

### PR DESCRIPTION
**Pull request by Pixep**
Will help mitigate CI tests errors, and we are actually unit testing the right classes

### CODE REVIEW

File: tests/generic/utils.py [hunk 1]
```diff
@@ -9,9 +9,9 @@ def fd_has_data(fd, timeout=0):
 
 
 class OsFile:
-    def __init__(self, fd, encoding):
+    def __init__(self, fd, encoding=None):
         self.fd = fd
-        self.encoding = encoding
+        self.encoding = encoding or 'ascii'
```
Comment (mjftw) [inline tests/generic/utils.py:14]: We might want to use UTF-8 here instead?

**Review by mjftw — APPROVED** [review]
This looks great!
Looks like we're heading towards not having any of those annoying false CI failures!

Only a minor comment, but approving anyway as it really is minor.

File: pluma/core/baseclasses/pexpectengine.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ def _on_opened(self):
 
     @property
     def is_open(self):
-        return self._pex and self._pex.isalive()
+        return (self._pex and self._pex.isalive()) is True
```
[the remark this answers is not in the document]
Comment (Pixep) [reply pluma/core/baseclasses/pexpectengine.py:35]: Got it, thanks. Done.

File: tests/generic/utils.py (hunk 1 above)
Comment (Pixep) [reply tests/generic/utils.py:14]: Probably yes, thought console engine default to ascii

**Pull request merged**